stdlib/public/SDK/Foundation/DataProtocol.swift

```diff
@@ -0,0 +1,239 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+#if os(macOS) || os(iOS)
+import Darwin
+#elseif os(Linux)
+import Glibc
+#endif
+
+// -- DataProtocol -------------------------------------------------------------
+public protocol DataProtocol : RandomAccessCollection where Element == UInt8, SubSequence : DataProtocol {
+    // -- Core Requirements ----------------------------------------------------
+    // FIXME: Remove in favor of opaque type on `regions`.
+    associatedtype RegionSequence: Sequence where RegionSequence.Element : DataProtocol & ContiguousCollection
```

Comment:
This is a pretty big FIXME that we're going to need to address. We might want to file a Radar to track this separately internally.

Reply:
(Filed 45739467)